app/Exception/HttpException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exception;
+
+use App\Enumerators\ResponseCode;
+use Throwable;
+
+class HttpException extends \Exception
+{
+    private ResponseCode $responseCode;
+
+    public function __construct(ResponseCode $responseCode, string $message = '', int $code = 0, Throwable $previous = null)
+    {
+        $this->responseCode = $responseCode;
+        parent::__construct($message, $responseCode->value, $previous);
+    }
+
+    public function getResponseHeader(): string
+    {
+        return 'HTTP/1.1 ' . $this->responseCode->value . ' ' . $this->message;
+    }
```

Comment:
Yeah this would become useless, also I think adding exception/error message to http status code header will result into some funky stuff.